driver/src/main/java/com/consol/citrus/db/driver/JdbcCallableStatement.java

```diff
@@ -31,106 +33,117 @@
 import java.sql.Ref;
 import java.sql.RowId;
 import java.sql.SQLException;
+import java.sql.SQLType;
 import java.sql.SQLXML;
 import java.sql.Time;
 import java.sql.Timestamp;
 import java.util.Calendar;
 import java.util.Map;
+import java.util.Objects;
 
 public class JdbcCallableStatement extends JdbcPreparedStatement implements CallableStatement {
 
-    public JdbcCallableStatement(final HttpClient httpClient, final String callableStatement, final String serverUrl, JdbcConnection connection) {
+    private Row resultRow;
+
+    public JdbcCallableStatement(final HttpClient httpClient,
+                                 final String callableStatement,
+                                 final String serverUrl,
+                                 final JdbcConnection connection) {
         super(httpClient, callableStatement, serverUrl, connection);
     }
 
     @Override
     public void registerOutParameter(final int parameterIndex, final int sqlType) throws SQLException {
-
+        setOutParameter(parameterIndex);
     }
 
     @Override
     public void registerOutParameter(final int parameterIndex, final int sqlType, final int scale) throws SQLException {
-
+        registerOutParameter(parameterIndex, sqlType);
     }
 
     @Override
     public boolean wasNull() throws SQLException {
-        return false;
+        final Row dataRow = getDataRow();
+        if(dataRow == null){
+            return false;
+        }
+        return Objects.isNull(dataRow.getLastValue());
     }
 
     @Override
     public String getString(final int parameterIndex) throws SQLException {
-        return null;
+        return (String) getDataRow().getValue(parameterIndex-1, String.class);
```

Comment:
I've just recognized that the driver was working with the `DataSet` object instead of the `JdbcResultSet` representation. I fixed that and this fixed the possible NPE as well.